server/server.js

```diff
@@ -293,6 +294,36 @@ app.get('/logout', (req, res) => {
     }
 });
 
+app.post('/flic_button_press', header(['button-serial-number','button-battery-level']).exists(), async (req, res) => {
+
+    try {
+        const validationErrors = validationResult(req)
+
+        if(validationErrors.isEmpty()){
+            let button = await db.getButtonWithSerialNumber(req.get('button-serial-number'))
+            if(button === null) {
+                log(`Bad request: Serial Number is not registered. Serial Number is ${req.get('button-serial-number')}`)
+                res.status(400).send();
+            }
+            else {
+                await db.saveButtonBatteryLevel(req.get('button-serial-number'), req.get('button-battery-level'))
+                await handleValidRequest(button, 1)
+                res.status(200).send()
+            }
+        }
+        else {
+            log(`Bad request, parameters missing ${validationErrors}`)
+            res.status(400).send()
+
+        }
+    }
+    catch(err) {
+        log(err)
+        res.status(500).send()
+    }
+});
+
+
 app.post('/', jsonBodyParser, async (req, res) => {
```

Comment:
Are we going to have an intermediate period where Buttons presses are sending messages to _both_ `/flic_button_press` and `/`?

Reply:
yes - the old fleet (Buchan, Viv, Kye7e, Princess Rooms?) are all doing the /